src/execution/collectFields.ts

```diff
@@ -26,14 +26,12 @@ import { typeFromAST } from '../utilities/typeFromAST.js';
 
 import { getDirectiveValues } from './values.js';
 
-export interface PatchFields {
-  label: string | undefined;
-  fields: Map<string, ReadonlyArray<FieldNode>>;
-}
+export type GroupedFieldSet = Map<string, ReadonlyArray<TaggedFieldNode>>;
 
-export interface FieldsAndPatches {
-  fields: Map<string, ReadonlyArray<FieldNode>>;
-  patches: Array<PatchFields>;
+export interface TaggedFieldNode {
+  fieldNode: FieldNode;
+  depth: number;
+  deferDepth: number | undefined;
```

Comment:
`depth` is just the depth we are at in terms of the overall operation, where root fields are at depth 0, their subfields are at depth 1, etc. It is used solely by `collectSubFields` to mark the `deferDepth`.

`deferDepth` is used to uniquely identify a the deferred fragment/payload that should include the given field node. As we merge defers at the same level, the depth can be used as a unique id.

Reply:
`depth` could really be hoisted out of the TaggedFieldNodes into a FieldSet interface as demonstrated by #3826 

`depth` will be the same for all TaggedFieldNodes, after all. However, this would mean that it would take an extra level of destructuring to get at the field nodes/deferDepth.

We can debate that within #3826 I suppose.

Reply:
why not never store the depth and just thread it through the execution algorithm and pass it as a parameter to collectSubFields?

Because then memoize3 would have a fourth parameter, that would always be the same for the given FieldSet...

Reply:
Going to mark this as resolved because was discussed at latest call, reference is at https://github.com/robrichard/graphql-spec/pull/10